packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -1353,28 +433,51 @@ export function CreateTaskForm({
   };
 
   const renderTaskDescriptionForm = () => {
-    if (taskRequest.category === 'patrol') {
-      return (
-        <PatrolTaskForm
-          taskDesc={taskRequest.description as PatrolTaskDescription}
-          patrolWaypoints={patrolWaypoints}
-          onChange={(desc) => handleTaskDescriptionChange('patrol', desc)}
-          allowSubmit={allowSubmit}
-        />
-      );
-    } else if (taskRequest.category === 'custom_compose') {
-      return (
-        <CustomComposeTaskForm
-          taskDesc={taskRequest.description as CustomComposeTaskDescription}
-          onChange={(desc) => {
-            handleCustomComposeTaskDescriptionChange(desc);
-          }}
-          allowSubmit={allowSubmit}
-        />
-      );
+    switch (taskRequest.category) {
+      case 'patrol':
+        return (
+          <PatrolTaskForm
+            taskDesc={taskRequest.description as PatrolTaskDescription}
+            patrolWaypoints={patrolWaypoints}
+            onChange={(desc) => handleTaskDescriptionChange('patrol', desc)}
+            allowSubmit={allowSubmit}
+          />
+        );
+      case 'delivery':
+        return (
+          <SimpleDeliveryTaskForm
+            taskDesc={taskRequest.description as SimpleDeliveryTaskDescription}
+            pickupPoints={pickupPoints}
+            dropoffPoints={dropoffPoints}
+            onChange={(desc) => handleTaskDescriptionChange('delivery', desc)}
+            allowSubmit={allowSubmit}
+          />
+        );
+      case 'custom_compose':
+        return (
+          <CustomComposeTaskForm
+            taskDesc={taskRequest.description as CustomComposeTaskDescription}
+            onChange={(desc) => {
+              handleCustomComposeTaskDescriptionChange(desc);
+            }}
+            allowSubmit={allowSubmit}
+          />
+        );
     }
 
     switch (taskRequest.description.category) {
+      case 'clean':
+        return (
+          <CleanTaskForm
+            taskDesc={taskRequest.description as CleanTaskDescription}
+            cleaningZones={cleaningZones}
+            onChange={(desc: CleanTaskDescription) => {
+              desc.category = taskRequest.description.category;
+              handleTaskDescriptionChange('compose', desc);
+            }}
+            allowSubmit={allowSubmit}
+          />
+        );
```

Comment:
That is outdated now, because we are now using perform_action to do a cleaning action, https://github.com/open-rmf/rmf_demos/blob/main/rmf_demos_tasks/rmf_demos_tasks/dispatch_clean.py